### ISSUE TITLE
Segmentation fault when too many GROUPID

### ISSUE THREAD

**Issue by bastiencarreres** [RickKessler/SNANA#1284]
Hi,
I think this error already happens before. I'm working on SNANA SIMLIB with HOST inputs for LSST. I've written a `OpSim` script that allow to add host to the generated SIMLIB file. It works well when generating a small number of LIBID (<1000). But when the number of LIBID increases (it's 50 000 by default) the number of GROUPID increase and launching SNANA result in seg fault.

I' m working on PERMULTER, SIMLIB and HOSTLIB files are in `/pscratch/sd/b/bastienc/LSST_SNANA/baseline_v3.2_10yrs`. The Input file is `/pscratch/sd/b/bastienc/LSST_SNANA/SNANALSST.input`.

If it's the cause of the crash, is it possible to increase considerably the number of groupid that we could put in SIMLIB?

**Comment by RickKessler** [comment]
more /pscratch/sd/b/bastienc/LSST_SNANA/SNANALSST.input
more: cannot open /pscratch/sd/b/bastienc/LSST_SNANA/SNANALSST.input: Permission denied

**Comment by RickKessler** [comment]
I found a few local string arrays that weren't large enough for the long GROUPID strings, so that fixed the crash. Next I enable GROUPID and increased MXGROUPD = 60 (was 20). Sim runs but quickly aborts with

 FATAL ERROR ABORT called by GEN_SNHOST_GALID
   Could not select HOSTLIB GALID  for ZGEN=0.0415 .
   This error is baffling ?!?!?!

so running valgrind to look for more array bound issues.

**Comment by bastiencarreres** [comment]
Thanks ! I reduced the number of host in the catalog for debug purpose so maybe it's why it doesn't found a convenient host?
However in this example there is only 10 000 LSST fields, I noticed that when I put the default number of 50 000 fields for the SIMLIB I could have few hundreds of GROUPID for one LIBID. Is it possible to push the MAXGROUPD number further away or is it better to found a better way to do that?
